src/raft.rs

```diff
@@ -742,22 +713,116 @@ impl<T: Storage> Raft<T> {
         }
     }
 
+    // recover configuration states to the latest one before `index`.
+    fn recover_conf_states_ahead(&mut self, index: u64) {
```

Comment:
Perhaps we could add some `debug!()` here? (Eg "Revert needed, reverting to conf state {:?}")